cartridge/remote-control.lua

```diff
@@ -327,6 +381,13 @@ local function rc_handle(s)
             break
         end
     end
+
+    -- Handler must wait for all tasks because socket, which is used by
+    -- them, will be closed forcefully after return. acquire_socket and
+    -- release_socket won't help.
+    for _, task in pairs(fiber.self().storage.tasks) do
+        task:join()
+    end
```

Comment:
Is it possible that new tasks will be spawned during `task:join()`?

Reply:
Nope. New tasks are spawned only in `communicate` and `task:join` is strictly after.